src/PSR7/ValidatorBuilder.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenAPIValidation\PSR7;
+
+use cebe\openapi\spec\OpenApi;
+use InvalidArgumentException;
+use OpenAPIValidation\PSR7\SchemaFactory\JsonFactory;
+use OpenAPIValidation\PSR7\SchemaFactory\PrecreatedSchemaFactory;
+use OpenAPIValidation\PSR7\SchemaFactory\YamlFactory;
+use OpenAPIValidation\PSR7\SchemaFactory\YamlFileFactory;
+use Psr\Cache\CacheItemPoolInterface;
+
+class ValidatorBuilder
+{
+    /** @var SchemaFactory */
+    protected $factory;
+    /** @var CacheItemPoolInterface */
+    protected $cache;
+    /** @var int|null */
+    protected $ttl;
+    /** @var string */
+    protected $cacheKey;
+
+    /**
+     * @return $this
+     */
+    public function setSchemaFactory(SchemaFactory $schemaFactory) : self
+    {
+        $this->factory = $schemaFactory;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function setCache(CacheItemPoolInterface $cache, ?int $ttl = null) : self
+    {
+        $this->cache = $cache;
+        $this->ttl   = $ttl;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function overrideCacheKey(string $cacheKey) : self
+    {
+        $this->cacheKey = $cacheKey;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function fromYaml(string $yaml) : self
+    {
+        $this->setSchemaFactory(new YamlFactory($yaml));
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function fromYamlFile(string $yamlFile) : self
+    {
+        $this->setSchemaFactory(new YamlFileFactory($yamlFile));
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function fromJson(string $json) : self
+    {
+        $this->setSchemaFactory(new JsonFactory($json));
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function fromJsonFile(string $jsonFile) : self
+    {
+        $this->setSchemaFactory(new JsonFactory($jsonFile));
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function fromSchema(OpenApi $schema) : self
```

Comment:
Ok, this method allows injecting any arbitrary schema without calling static factory methods like `fromYaml`. Can you elaborate usecases when this is used. Just to make sure I follow the logic.

Reply:
Can we chain this with cache somehow?

Reply:
I hope while calling `fromSchema` - cache is handled by callee somewhere else, but built-in cache would work if you set cache and override cachekey for it (but would be useless tging for real).

Generic usecase is to use schemase generated by hand by manually constructing `cebe/OpenApi`  from objects.

Reply:
It makes total sense! Can you add a couple of words in the Readme about this option?

Reply:
Sure